### ISSUE TITLE
expanded keyword paremeter without default value doesn't refer proper variable.

### ISSUE THREAD

**Issue by ktakashi** [ktakashi/sagittarius-scheme#21]
**[Original report](https://bitbucket.org/ktakashi/sagittarius-scheme/issues/16) by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

The following script should print keyword and value pair and return ok but got unbound variable error.

```
#!scheme
(import (rnrs))

(define-syntax define-key
  (lambda (x)
    (define (key&name sym) (list (make-keyword sym) sym))
    (define (params slots)
      (let loop ((slots slots) (r '()))
	(syntax-case slots ()
	  (() (reverse! r))
	  ((name . rest)
	   (loop (cdr slots) (cons (key&name (syntax->datum #'name)) r))))))
    (syntax-case x ()
      ((k name (p ...) body ...)
       (with-syntax ((((keys names) ...) 
		      (datum->syntax #'k (params #'(p ...)))))
	 #'(define (name :key names ...)
	     (print (apply append! (list `(keys ,names) ...)))
	     body ...))))))

(define-key test (a b c) 'ok)
(test)
```

**Comment by ktakashi** [comment]
**Original comment by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

This works if I modify it like this.
```
#!scheme
(define (name :key (names #f) ...) ...)
```

**Comment by ktakashi** [comment]
**Original comment by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

Fixed not to use unwrapped variable for let-keyword* (fixes #16).
